spec/core/connector_settings_spec.rb

```diff
@@ -62,6 +62,153 @@
     end
   end
 
+  context '.filtering' do
+    context 'filtering is present' do
+      let(:elasticsearch_response) {
+        {
+          :filtering => [
+            {
+              :domain => 'DEFAULT',
+              :active => {
+                :active_rules => [],
+                :advanced_config => {},
+              }
+            }
+          ]
+        }
+      }
+
+      it 'extracts filtering field' do
+        filtering = subject.filtering
+        first_filter = filtering[0]
+
+        expect(first_filter[:domain]).to eq('DEFAULT')
+        expect(first_filter[:active][:active_rules]).to_not be_nil
+        expect(first_filter[:active][:advanced_config]).to_not be_nil
+      end
+    end
+
+    context 'filtering is not present' do
+      it 'returns default filtering object' do
+        filtering = subject.filtering
+
+        expect(filtering).to_not be_nil
+        expect(filtering.empty?).to eq(true)
```

Comment:
```suggestion
        expect(filtering).to be_empty
```

nit: this is slightly more idiomatic. See:

https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/predicate-matchers